journal/src/main/scala/com/evolutiongaming/kafka/journal/Journal.scala

```diff
@@ -343,4 +343,17 @@ object Journal {
 
     implicit val configReaderConsumerPoolConfig: ConfigReader[ConsumerPoolConfig] = deriveReader
   }
+
+  trait DataIntegrityConfig {
+    def seqNrUniqueness: Boolean
+  }
+
+  object DataIntegrityConfig {
+
+    private case class Implementation(seqNrUniqueness: Boolean) extends DataIntegrityConfig 
+
+    val Default: DataIntegrityConfig = Implementation(seqNrUniqueness = true)
+
+    implicit val configReaderDataIntegrityConfig: ConfigReader[DataIntegrityConfig] = deriveReader[Implementation].map(a => a: DataIntegrityConfig)
+  }
```

Comment:
Is `seqNrUniqueness: Boolean` the best we can do? Nicely named ADTs might be more expressive: `CheckSeqNrGrowing` (default) and `IgnoreChecks` (new) 

Frankly, we never check for `Uniqueness` - we check that new `seqNr` larger than previously observed one, but we never cehck for delta to be exactly one, do we? We could add `StrictCheckSeqNr` to do checks for exactly `+1` though...

Reply:
I'm fine with naming and strongly against ADT for true|false values, please do not increase complexity without reason. 

> Frankly, we never check for Uniqueness - we check that new seqNr larger than previously observed one, but we never cehck for delta to be exactly one, do we? We could add StrictCheckSeqNr to do checks for exactly +1 though...

KJ should not check that current `seqNr` is exactly +1 of previous due to possible gaps in the sequence. Now final decision of accepting or rejecting event delegated to the application instead of been forced in KJ.